subsys/net/l2/ethernet/ethernet.c

```diff
@@ -402,7 +402,10 @@ static enum net_verdict ethernet_recv(struct net_if *iface,
 		return net_gptp_recv(iface, pkt);
 	}
 
-	ethernet_update_length(iface, pkt);
+	if (type != NET_ETH_PTYPE_EAPOL)
```

Comment:
Could you add in the commit
`net: ethernet: Fix EAPol packet length changed wrongly`
a bit more information that answers the question "why" this change is needed.
So something like
```
In ethernet_recv(), for EAPol packet, its length should not be modified
like other IP packets because ....
```
or something like this. Now it is not obvious why this is needed.

Reply:
will update